solv/tdnfpackage.c

```diff
@@ -820,6 +820,58 @@ SolvGetPackageId(
     goto cleanup;
 }
 
+uint32_t
+SolvGetPkgChecksumFromId(
+    PSolvSack pSack,
+    uint32_t dwPkgId,
+    char** ppszChecksum)
+{
+
+    uint32_t dwError = 0;
+    const char* pszTemp = NULL;
+    char* pszChecksum = NULL;
+    Solvable *pSolv = NULL;
+    int checksumtype= REPOKEY_TYPE_SHA256;
```

Comment:
Can you find out if repo data can use other types than sha256?

Reply:
Look at https://github.com/rpm-software-management/createrepo_c , it looks like it supports different hash types:

checksum | Checksum type to use | md5, sha, sha1, sha224, sha256, sha384, sha512 | sha256